File: JAVA/3194. Minimum Average of Smallest and Largest Elements.java
class Solution {
    public double minimumAverage(int[] n) {
        Arrays.sort(n);
        double m=9999.9;
        for(int i=0;i<n.length/2;i++){
            m=Math.min(m,(n[i]+n[n.length-1-i])/2.0);
        }
        return m;
    }
}

File: C++/3194. Minimum Average of Smallest and Largest Elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& n) {
        double m=9999.9;
        sort(n.begin(),n.end());
        for(int i=0;i<n.size()/2;i++){
            m=min(m,(n[i]+n[n.size()-1-i])/2.0);
        }
        return m;
    }
};